From bf973ddd2ec5b5555fd03e1aac6fe6e4d1e60642 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Fri, 15 Jul 2022 12:04:48 +0200
Subject: [PATCH] setup-disk: fix hang with UEFI and interactive 'none'

Fix a hang that happens with interactively selecting 'none' with efi
mode.

Fixes commit 851247d96340 (setup-disk: calculate size of ESP parition
for grub)
---
 setup-disk.in | 13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

diff --git a/setup-disk.in b/setup-disk.in
index 0f00de9..c479f97 100644
--- a/setup-disk.in
+++ b/setup-disk.in
@@ -1545,13 +1545,12 @@ if [ $# -gt 0 ]; then
 else
 	ask_disk "Which disk(s) would you like to use? (or '?' for help or 'none')" \
 		diskselect_help $disks
-	if [ "$resp" != none ]; then
-		for i in $resp; do
-			diskdevs="$diskdevs /dev/$i"
-		done
-	else
-		DISK_MODE="none"
+	if [ "$resp" = "none" ]; then
+		exit 0
 	fi
+	for i in $resp; do
+		diskdevs="$diskdevs /dev/$i"
+	done
 fi
 
 if [ -n "$diskdevs" ] && [ -z "$DISK_MODE" ]; then
@@ -1616,7 +1615,7 @@ if [ $# -gt 1 ]; then
 	USE_RAID=1
 fi
 
-dmesg -n1
+$MOCK dmesg -n1
 
 if [ -n "$USE_LVM" ] && ! grep -w -q device-mapper /proc/misc; then
 	modprobe dm-mod
-- 
2.37.1

